code/modules/assembly/signaler.dm

```diff
@@ -190,8 +191,35 @@
 	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
 	resistance_flags = FIRE_PROOF
+	eaten_text = "you've ate the anomaly core..."
 	var/anomaly_type = /obj/effect/anomaly
 
+/obj/item/assembly/signaler/anomaly/attack(mob/living/M, mob/living/user)
+	if(user.a_intent == INTENT_HARM || M != user || !ishuman(user))
+		return ..()
+
+	var/mob/living/carbon/human/H = user
+	var/obj/item/organ/stomach/S = H.getorganslot(ORGAN_SLOT_STOMACH)
+
+	if(!istype(S, /obj/item/organ/stomach/cell))//need a fancy stomach for it
+		return ..()
+
+	if(!eaten(H))
+		return ..(src)
```

Comment:
Since the proc parameters are `(mob/living/M, mob/living/user)` this will break

Reply:
checked this seems to be running fine, gotta ask how this will break it. im not a master Prestidge coder and i cant look at something and see the problem until it breaks in vs code or it does something weird during testing. telling me that it will break without a reason doesnt help improve what im trying to do, with reason then i can go on from that knowing better.